zerver/lib/actions.py

```diff
@@ -976,13 +976,6 @@ def render_incoming_message(message: Message,
     return rendered_content
 
 def get_typing_user_profiles(recipient: Recipient, sender_id: int) -> List[UserProfile]:
-    if recipient.type == Recipient.STREAM:
-        '''
-        We don't support typing indicators for streams because they
-        are expensive and initial user feedback was they were too
-        distracting.
-        '''
-        raise ValueError('Typing indicators not supported for streams')
```

Comment:
This block should be replaced with an appropriate bulk query; we have some nice code for doing that in the message sending code path that you'll want to borrow or call.